scripts/src/java/org/oppia/android/scripts/testfile/TestFileCheck.kt

```diff
@@ -23,44 +23,79 @@ fun main(vararg args: String) {
 
   val testFileExemptiontextProto = "scripts/assets/test_file_exemptions"
 
-  // A list of all the files to be exempted for this check.
-  // TODO(#3436): Develop a mechanism for permanently exempting files which do not ever need tests.
-  val testFileExemptionList = loadTestFileExemptionsProto(testFileExemptiontextProto)
-    .getExemptedFilePathList()
-
-  // A list of all kotlin files in the repo to be analyzed.
-  val searchFiles = RepositoryFile.collectSearchFiles(
-    repoPath = repoPath,
-    expectedExtension = ".kt",
-    exemptionsList = testFileExemptionList
-  )
-
-  // A list of all the prod files present in the repo.
-  val prodFilesList = searchFiles.filter { file -> !file.name.endsWith("Test.kt") }
-
-  // A list of all the test files present in the repo.
-  val testFilesList = searchFiles.filter { file -> file.name.endsWith("Test.kt") }
-
-  // A list of all the prod files that do not have a corresponding test file.
-  val matchedFiles = prodFilesList.filter { prodFile ->
-    !testFilesList.any { testFile ->
-      testFile.name == computeExpectedTestFileName(prodFile)
-    }
+  val testFileCheck = TestFileCheck(repoPath, testFileExemptiontextProto)
+  testFileCheck.execute()
+}
+
+/**
+ * Class for checking the presence of test files in a repository.
+ *
+ * @param repoPath The path of the repo
+ * @param testFileExemptiontextProto the location of the test file exemption textproto file.
+ */
+class TestFileCheck(
+  private val repoPath: String,
+  private val testFileExemptiontextProto: String
+) {
+
+  /**
+   * Retrieves a list of files exempted from requiring a test file based on the specified protocol buffer.
+   *
+   * This function loads the test file exemptions from the provided protocol buffer file path,
+   * filters and returns list of paths for files that are not required for having a test file.
+   *
+   * The files specifically marked as exempt from needing a test file are included,
+   * rather than those exempt from meeting a minimum coverage percentage.
+   *
+   * @param testFileExemptiontextProto the location of the test file exemption textproto file.
+   * @return A list of file paths that are exempted from requiring a test file.
+   */
+  fun getTestFileExemptionList(testFileExemptiontextProto: String): List<String> {
```

Comment:
Why is this exposed? Generally, prefer never exposing methods outside the scope of the class.

If it's because it's going to be needed in the coverage scripts, I suggest just duplicating it there since the load logic is quite straightforward (plus we use utilities when sharing functionality between scripts, we usually don't have the scripts be cross-dependent).

Reply:
Hhmm... Okay! Again my bad🥲! So for all the time I thought, the **searchFiles = RepositoryFile.collectSearchFiles()** is actually retrieving the original app's repository files and that is why I was trying to create the seperation,
**ie**. to not touch the searchFiles, as the original sources have a seperate script_exemption.textproto list to handle the test file check situation which is different from the custom **test_exemption.pb** list that I provided to it.

So, I didn't have a proper realization that it was actually only searching within the given **repoPath**,
```
val searchFiles = RepositoryFile.collectSearchFiles(
      repoPath = repoPath,
      expectedExtension = ".kt",
      exemptionsList = testFileExemptionList
    )
```
that single thing just clears up everything for me and just realized it is just a simple call to the execute(), and me just complicated the tests.

Sorry for that, I re-wrote the tests with a success and an error case. Hope this provides proper testing to the actual end user behaviour.

Reply:
So one another thing, is the original source has KDocs for the private methods - just wondering if they have separate standards / practice or is that something to remove or can I just ignore them as it is not related : https://github.com/oppia/oppia-android/blob/dcde7b359e3f3ba63f50f7585b45ff6ef6b12440/scripts/src/java/org/oppia/android/scripts/testfile/TestFileCheck.kt#L67-L97

Reply:
Good question--it's fine to remove it (we shouldn't have added it originally, but it was missed during review). However, it's often easier just to maintain existing code and only clean it up if we're already making changes to it. Since this PR is moving code, either outcome seems reasonable.